lib/observer.js

```diff
@@ -231,7 +231,16 @@ ObserverMixin.notifyObserversAround = function(operation, context, fn, callback)
 
       function cbForWork(err) {
         const args = [].slice.call(arguments, 0);
-        if (err) return callback.apply(null, args);
+        if (err) {
+          // call observer in case of error to hook response
+          err.status = 500;
+          context.res = {err};
```

Comment:
We should probably omit out `err.status = 500` and let the handler decide what to do with the error.

And the error object should be contained in `context.error` instead of `context.res`.

@bajtos @raymondfeng, thoughts?

Reply:
Yes please, it should be `context.error`.